### PR TITLE
Matt/opentype features

### PR DESCRIPTION
**Pull request by mattysteves**
### Changes

- opentype.js library added to dependencies
- Added detection of opentype features from gsub table of font file to App.tsx.
- When a font is dropped, opentype features of the font are logged to the console
- When a font is dropped, opentype features are displayed as text underneath the line height div (not final).

### Tests applied

Tested with otf fonts with known features. Features were recorded correctly. Fonts with no features return an empty list.

### Issue ticket number(s)

- [#73 ](https://github.com/typefaceoff/typefaceoff/issues/73)

### Checklist

- [ ] Documented
- [ ] Linting tests successful
- [ ] merge updated prod branch into development branch

### CODE REVIEW

**Review by SamharAeron — CHANGES_REQUESTED** [review]
Please resolve merge conflicts first :)

File: src/components/App.tsx [hunk 1]
```diff
@@ -17,14 +18,48 @@ function App() {
...
+  // Opentype feature option names from the gsub table of the font file on the left
+  const [fontFeatureOptionsLeft, setFontFeatureOptionsLeft] = useState<unknown[]>([]);
+
+  // Opentype feature option names from the gsub table of the font file on the right
+  const [fontFeatureOptionsRight, setFontFeatureOptionsRight] = useState<unknown[]>([]);
+
   // Handler for when a font is selected on the left side
   const handleFontSelectedLeft = (selectedFont: File | null) => {
     setSelectedFontLeft(selectedFont);
+    if (selectedFont != null) {
```
Comment (jess-lowe) [inline src/components/App.tsx:33]: A bulk of this logic could be extracted away into another method as it essentially is the same for left and right but for now, it looks pretty good!

File: src/components/App.tsx [hunk 2]
```diff
@@ -20,14 +21,48 @@ function App() {
...
   const handleFontSelectedRight = (selectedFont: File | null) => {
     setSelectedFontRight(selectedFont);
+    if (selectedFont != null) {
+      const buffer = selectedFont.arrayBuffer();
+      buffer.then((data) => {
+        const otfFont = opentype.parse(data);
+        const featureNames = [
+          ...Array.from(new Set(otfFont.tables.gsub.features.map((f: any) => f.tag))),
+        ];
+        //TODO: Remove debug logging of feature names
```
Comment (mwan391) [inline src/components/App.tsx]: Buffer implementation looking good. 
Will these TODO be done in a separate PR? Can they be removed here?

**Review by mwan391 — CHANGES_REQUESTED** [review]

File: src/components/App.tsx (hunk 2 above)
Comment (mattysteves) [reply src/components/App.tsx]: Done!

File: src/components/App.tsx (hunk 1 above)
Comment (SamharAeron) [reply src/components/App.tsx:33]: I agree, if this is pulled through then a refactor would be needed.

**Review by SamharAeron — COMMENTED** [review]
Once the merge conflicts and build issues are resolved this looks good, prefer if the shared logic can be abstracted to separate method but if not done in this pr then a refactor issue needs to be created.

**Comment by mattysteves** [comment]
Closing PR, problems locally with lock-file are preventing merge. Opted for a new pr with repaired lock-files.
